src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -279,22 +285,32 @@ private void UpdateCurrentConfiguration()
             {
                 Interlocked.Exchange(ref _configurationRetrieverState, ConfigurationRetrieverIdle);
             }
-#pragma warning restore CA1031 // Do not catch general exception types
         }
 
+        /// <summary>
+        /// Called only when configuration is successfully obtained.
+        /// </summary>
+        /// <param name="configuration"></param>
         private void UpdateConfiguration(T configuration)
         {
             _currentConfiguration = configuration;
-            _syncAfter = DateTimeUtil.Add(DateTime.UtcNow, AutomaticRefreshInterval +
-                TimeSpan.FromSeconds(new Random().Next((int)AutomaticRefreshInterval.TotalSeconds / 20)));
+            // StartupTime is the time when ConfigurationManager was instantiated.
+            // (DateTimeOffset.UtcNow - StartupTime).TotalSeconds is the number of seconds since ConfigurationManager was instantiated.
+            // For automatic refresh, we add a 5% jit.
+            // Record in seconds when the last time configuration was obtained.
+            double secondsWhenRefreshOccurred = (DateTimeOffset.UtcNow - StartupTime).TotalSeconds +
+                                 ((_automaticRefreshIntervalInSeconds >= int.MaxValue) ? 0 : (new Random().Next((int)_automaticRefreshIntervalInSeconds / 20)));
```

Comment:
`int.MaxValue` seconds is nearly 600000 hours. This doesn't seem like a reasonable value for `_automaticRefreshIntervalInSeconds`, so I think when it's set there could be a check like this that rejects unrealistically large values. Then you don't need this check.

Also, you can precompute `_automaticRefreshIntervalInSeconds / 20` in the setter and store it in a value called something like `_maxJitter`.